### DISCUSSION TITLE
Font Properties

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/GoOdf#53] [General]
I tried setting font to Arial 9pt Bold and the text reverts to normal. 

Besides, this window keeps popping up:
![image](https://github.com/larspalo/GOODF/assets/6999990/914e23d3-0bf3-4438-b749-5fb66d5703a0)

It is a bit of a nuisance. Why ask to revert if I am specifically making a change intentionally?

**Comment by larspalo** [comment]
In https://github.com/larspalo/GOODF/discussions/13 I had a discussion with davidgritter about fonts. In https://github.com/larspalo/GOODF/discussions/13#discussioncomment-5572440 we implemented current behaviour. However I did notice a slightly strange behaviour when testing randomly changing font faces and sizes, so this is definitely something I'll look into! I believe the original intention was to ask if font (and possibly size) should revert if cancel was pressed - as that method is mostly used otherwise in GOODF to empty a value.

However, for your information, as mentioned in the discussion quoted above, GrandOrgue cannot really read/use the extra information about a font - thus it's meaningless to add bold, italic or condensed or whatever extra font information. Just the plain font name (font face) is actually used. GOODF does exactly the same thing as GrandOrgue when writing the odf, which unfortunately also means that it's possible to select fonts with extra decorations in the GUI that actually won't be used at all (only the plain version will be). I've not found a way to limit what fonts are reported by the font selection dialog that's triggered from the font picker, it just reports everything that is present on the users system.

**Reply by Bumblebee001** [reply]
![image](https://github.com/larspalo/GOODF/assets/6999990/f0d8c2b6-b515-424a-b694-b5696173fdf2)

This is in GO (not GOODF)

Arial 11pt Black (not the colour)

**Comment by larspalo** [comment]
Yes, Arial Black seems to be a separate face name whereas bold, italic, condensed versions are not. GOODF however does the same thing as GO does in regards to fonts so if it's written from GOODF it can also be used in GO. Remember though that it's not always certain that a specific font face will be present on another users computer.

The dialog shown above has indeed some buggy behaviour though, I'll look into it eventually!

**Reply by Bumblebee001** [reply]
This is odd..... The "black" is listed together with bold, italics etc - not as a separate font / type face.

**Reply by larspalo** [reply]
Yes, it's definitely a source of confusion. There's both an "Arial Black" and an "Arial Black Regular". Even with my most recent commit there's a strange behaviour caused by that, I'll have to test and work on it some more to make sure the fonts are read and written correctly each time.
